qiskit/primitives/base_sampler.py

```diff
@@ -359,6 +359,13 @@ def run(
                     f"the number of parameters ({circuit.num_parameters}) for the {i}-th circuit."
                 )
 
+        for i, circuit in enumerate(circuits):
+            if circuit.num_clbits == 0:
+                raise QiskitError(
+                    f"The {i}-th circuit does not have any classical bit. "
+                    "Sampler requires classical bits."
```

Comment:
Maybe we can add some more information here and tell users they also should add measurements (otherwise one could think just adding a clbit would be enough 🙂 )
```suggestion
                    f"The {i}-th circuit does not have any classical bit. "
                    "Sampler requires classical bits, plus measurements "
                    "on the desired qubits."
```

Reply:
As for measurement, I copied another check from Sampler reference implementation.
https://github.com/Qiskit/qiskit-terra/blob/a4cf1483c8ccf711c9b01a70288d1a35affd4693/qiskit/primitives/sampler.py#L159